extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -88,19 +82,20 @@ void build(BuildProducer<AdditionalBeanBuildItem> additionalBeanProducer,
                 .collect(Collectors.toSet());
         new FlywayDatasourceBeanGenerator(dataSourceNames, generatedBeanBuildItem).createFlywayProducerBean();
 
-        registerNativeImageResources(resourceProducer, generatedResourceProducer,
-                discoverApplicationMigrations(getMigrationLocations(dataSourceNames)));
+        List<String> applicationMigrations = discoverApplicationMigrations(getMigrationLocations(dataSourceNames));
+        recorder.setApplicationMigrationFiles(applicationMigrations);
 
-        containerListenerProducer.produce(
-                new BeanContainerListenerBuildItem(recorder.setFlywayBuildConfig(flywayBuildConfig)));
+        resourceProducer.produce(new NativeImageResourceBuildItem(applicationMigrations.toArray(new String[0])));
+
+        containerListenerProducer.produce(new BeanContainerListenerBuildItem(recorder.setFlywayBuildConfig(flywayBuildConfig)));
     }
 
     /**
      * Handles all the operations that can be recorded in the RUNTIME_INIT execution time phase
      *
      * @param recorder Used to set the runtime config
      * @param flywayRuntimeConfig The Flyway configuration
-     * @param dataSourceInitializedBuildItem Added this dependency to be sure that Agroal is initialized first
+     * @param jdbcDataSourceBuildItems Added this dependency to be sure that Agroal is initialized first
```

Comment:
Could you clean up all these useless comments? I did it in the Liquibase extension that was based on it. It should be done in a separate commit.

Reply:
Done.